app/elements/common/settings/settings-page.html

```diff
@@ -41,21 +42,27 @@
 		<user-service id="userService" user="{{user}}"></user-service>
 		<paper-material id="wrapper" elevation="1" class="card">
 			<div class="mainContent">
-				<div id="fieldList">
+				<form is="iron-form" id="form">
 					<div>
 						<div class="list-item">
 							<div class="item-label">Name</div>
 							<div>{{fullName}}</div>
 						</div>
 						<div class="aligned-item">
-							<paper-input label="First name" value="{{user.firstName}}"></paper-input>
-							<paper-input label="Last name" value="{{user.lastName}}"></paper-input>
+							<paper-input label="First name" value="{{user.firstName}}"
+								pattern="[a-zA-Z]+" minlength="5" maxlength="40" auto-validate
+								error-message="Must be between 5 and 40 chars and only contain letters."></paper-input>
+							<paper-input label="Last name" value="{{user.lastName}}"
+								pattern="[a-zA-Z]+" minlength="5" maxlength="40" auto-validate
+								error-message="Must be between 5 and 40 chars and only contain letters."></paper-input>
 						</div>
 					</div>
 					<div>
 						<div class="list-item">
 							<div class="item-label">Username</div>
-							<paper-input value="{{user.username}}" no-label-float></paper-input>
+							<paper-input value="{{user.username}}" no-label-float
+								required minlength="5" maxlength="40" auto-validate pattern="^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$"
```

Comment:
THis regex allows for a maximum of one dash ("-") in the username. What is the point here?